### DISCUSSION TITLE
Nothing shows & No Error

### DISCUSSION THREAD

**Discussion by rwillox** [ruffle-rs/ruffle#5812] [Q&A] [answered]
Using the basic web script (ruffle.rs in same folder) nothing displays with no error reported just a blank screen:-
`<script>
    window.RufflePlayer = window.RufflePlayer || {};
    window.addEventListener("load", (event) => {
        const ruffle = window.RufflePlayer.newest();
        const player = ruffle.createPlayer();
        const container = document.getElementById("container");
        container.appendChild(player);
        player.load("Heritage.swf").then(() => {
        console.info("Ruffle successfully loaded the file");
        }).catch((e) => {
        console.error(`Ruffle failed to load the file: ${e}`);
        });
</script>
<script src="ruffle.js"></script>`

Apache Version | 2.4.51
New user/first attempt so not sure where to look first!

regards

**Comment by adrian17** [comment]
Do you see any errors/indicators in the console and network request list?

**Comment by rwillox** [comment]
Thanks for quick response! As said, completely new user (downloaded
yesterday) and first try following the docs. 

This is the page link:-

https://www.bo-ness.org.uk/trails/index.html

 

This is from Console:-

index.html:30 Uncaught TypeError: Cannot read properties of null (reading
'appendChild')

    at index.html:30

VM397 ruffle.js:1 Couldn't load Ruffle into
IFRAME[chrome-extension://noojglkidnpfjbincgijbaiedldjfbhh/data/shared/tpc-c
heck.html]: SecurityError: Blocked a frame with origin
"https://www.bo-ness.org.uk" from accessing a cross-origin frame.

Se @ VM397 ruffle.js:1

DevTools failed to load source map: Could not load content for
chrome-extension://eakacpaijcpapndcfffdgphdiccmpknp/scripts/content_page.map
: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME

DevTools failed to load source map: Could not load content for
chrome-extension://liecbddmkiiihnedobmlmillhodjkdmb/js/content.js.map: HTTP
error: status code 404, net::ERR_UNKNOWN_URL_SCHEME

DevTools failed to load source map: Could not load content for
chrome-extension://liecbddmkiiihnedobmlmillhodjkdmb/js/companion-bubble.js.m
ap: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME

DevTools failed to load source map: Could not load content for
chrome-extension://liecbddmkiiihnedobmlmillhodjkdmb/js/intercom-link-expand-
loader.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME

DevTools failed to load source map: Could not load content for
https://www.bo-ness.org.uk/trails/ruffle.js.map: HTTP error: status code
404, net::ERR_HTTP_RESPONSE_CODE_FAILURE

 

Also, added the suggested mime type line to in-folder htaccess in case that
was the problem although no error was given but made no difference.

 

regards

**Comment by rwillox** [comment]
Hi

 

Just tried an original file that used to load (flash.htm) with the Plug &
Play option and get a wasm error! Info:-

# Error Info

Error name: TypeError

Error message: Failed to fetch

Error stack:

```

TypeError: Failed to fetch

    at r (https://bo-ness.org.uk/trails/ruffle.js:1:36540)

    at https://bo-ness.org.uk/trails/ruffle.js:1:38988

    at q (https://bo-ness.org.uk/trails/ruffle.js:1:38994)

    at HTMLElement.ensureFreshInstance
(https://bo-ness.org.uk/trails/ruffle.js:1:51563)

    at HTMLElement.load (https://bo-ness.org.uk/trails/ruffle.js:1:53936)

    at HTMLElement.connectedCallback
(https://bo-ness.org.uk/trails/ruffle.js:1:71135)

    at Ee (https://bo-ness.org.uk/trails/ruffle.js:1:76404)

    at https://bo-ness.org.uk/trails/ruffle.js:1:82063

    at Object.polyfill (https://bo-ness.org.uk/trails/ruffle.js:1:82202)

    at Re.init (https://bo-ness.org.uk/trails/ruffle.js:1:81138)

```

 

# Player Info

Allows script access: true

 

# Page Info

Page URL: https://bo-ness.org.uk/trails/flash.htm

 

# Browser Info

Useragent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36
(KHTML, like Gecko) Chrome/96.0.4664.45 Safari/537.36

OS: Win32

 

# Ruffle Info

Version: 0.1.0

Name: nightly 2021-12-03

Channel: nightly

Built: 2021-12-03T00:11:19.900Z

Commit: 54fcfde1cc7a5ba29024857a10c1cf8c7173d0f2

Is extension: false

 

Don't have access to server config.

 

regards

**Comment by adrian17** [comment]
> index.html:30 Uncaught TypeError: Cannot read properties of null (reading
'appendChild')

Indeed, the code you have on the page embeds player on the element with ID "container", but the side doesn't have such an element.

**Comment by rwillox** [comment, marked as answer]
Hi Adrian

 

Seem to have solved it and now working! Made an original mistake and only
uploaded ruffle.js but when uploaded all the files and tried the
old/original embed file, now seems to load as expected.

 

Thanks for your quick response and  help.

 

Regards

Rob Willox
